print_docs.py

```diff
@@ -299,11 +333,26 @@ def linkify_type(string):
     splitstr = re.split(r'([\s\[\]\(\)\{\}])', string)
     tks = map(lambda s: linkify(s, loc_map), splitstr)
     return "".join(tks)
-
+  def linkify_named_ref(name, string):
+    if string in bib.entries:
```

Comment:
It'd be nice to generate backrefs, i.e. links on `references.html` back to the pages where the citations occur, but I need a way to get the file name here. 

I don't have much experience with jinja templates, so I was thinking of changing `convert_markdown` to a higher order function like this:

```python
def convert_markdown(filename):
  def _convert(string):
    # current body of convert_markdown
    pass
  return _convert
```

And then I'd change all the templates using `convert_markdown` to pass their filenames as an argument. Is there a cleaner way of doing this?

Reply:
I don't know about clean, but I'd rather go for global variables.  You'll also need a way to generate ids for every link.

Reply:
What did you have in mind with global variables? Do you mean having something like `current_file` in the global scope and then assigning to it each time we start rendering another file?

Reply:
OK, got it! For the backrefs, I was thinking of just using a new field for each item in `bib.entries` (which is already global).